model/Aktivita/Program.php

```diff
@@ -537,7 +537,7 @@ private function tiskObsahuTabulky(?array &$aktivitaRaw, $denId = null)
         }
 
         if ($aktivit == 0) {
-            $sloupcu = PROGRAM_KONEC - PROGRAM_ZACATEK + 1;
+            $sloupcu = count(Program::seznamHodinZacatku());
```

Comment:
Nechybí nám tu to pojistná +1?

Ale popravdě nevím k čemu původně byla...

Reply:
Vycházelo mi to bez ní právěže :grin:

Reply:
No každopádně to nedává ekvivalentní výsledky s předchozí verzí. Takže jestli tam tam **+1** nebyla jen kvůli nějaké následující chybě která byla teď opravená tak je to špatně.
```PHP
// PROGRAM_KONEC = 1
// PROGRAM_ZACATEK = 0

// předchozí verze
$sloupcu = PROGRAM_KONEC - PROGRAM_ZACATEK + 1;
$sloupcu = 1 - 0 + 1;
$sloupcu = 2;

// současná verze
$hodinyZacatku = range(PROGRAM_ZACATEK, PROGRAM_KONEC - 1, 1);
$hodinyZacatku = range(0, 1 - 1, 1);
$hodinyZacatku = range(0, 0, 1);
// PHP má inkluzivní horní hranici range
$hodinyZacatku = [0];
$sloupcu = count($hodinyZacatku);
$sloupcu = count([0]);
$sloupcu = 1;
```

Reply:
@BurgrLada @Sciator Tak já to rozseknu.

Ono se to používá pouze na šířku buňky v případě, že ten den není ani jedna aktivita.

Když není, vypíše se text _Žádné aktivity tento den_. A šířka buňky (colspan) má jediný úkol, aby to nezmáčklo text, ale aby ho pěkně roztáhlo do více buněk.

Takže to vyřeším tímhle historickým, ale stále funkčním hackem
```html
    <td colspan="100%">
```
a přesný colspan pošleme k šípku.